### PR TITLE
Ruby: Support for LRO tests

### PR DESCRIPTION
**Pull request by landrito**

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/ruby/test.snip [hunk 1]
```diff
@@ -122,6 +123,39 @@
...
+    @end
+    @# Create expected grpc response
+    {@initCode(test.mockResponse.initCode)}
+    result = Google::Protobuf::Any.new
+    result.pack(expected_response)
+    operation = Google::Longrunning::Operation.new(
+      name: 'operations/{@test.name}',
+      done: true,
+      response: result
+    )
```
Comment (swcloud) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Can we add a test case where the result is an 'error'?

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Hm. That sounds feasible let me work on it.

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Done.

File: src/main/resources/com/google/api/codegen/ruby/test.snip [hunk 2]
```diff
@@ -122,6 +123,68 @@
...
+
+    {@test.grpcStubCallString}.stub(:new, mock_stub) do
+      client = {@test.serviceConstructorName}.new
+
+      @# Call method
+      {@methodCallWithResponse(test)}
+
+      @# Verify the response
+      assert(response.error?)
+      assert_equal(error, response.error)
```
Comment (swcloud) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Where does the 'error' here come from? I think it should be custom_error.

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Ah realized the error must also be of type `Google::Rpc::Status`. Fixed.

**Pull request merged**